src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -574,21 +574,12 @@ public function create_item( $request ) {
 			);
 		}
 
-		// Do not allow comments to be created with a non-default type.
-		if ( ! empty( $request['type'] ) && 'comment' !== $request['type'] ) {
-			return new WP_Error(
-				'rest_invalid_comment_type',
-				__( 'Cannot create a comment with that type.' ),
-				array( 'status' => 400 )
-			);
-		}
-
```

Comment:
I don't think this should be removed.

As the change to the unit tests show, it allows the addition of comment types to the database that WordPress is unable to support. This is problematic for a number of reasons, primarily because it fills the databse with junk, secondly because plugins may not expect the data and throw a fatal error as a result.